recipes/configs/qat.yaml

```diff
@@ -0,0 +1,78 @@
+# Config for multi-device QAT finetuning in qat_distributed.py
```

Comment:
Sounds good. I mostly just followed `quantization.yaml`, `eleuther_evaluation.yaml`, and `generation.yaml` so far in this PR. I think these configs have the same problem. Both (2) and (3) make sense to me. What do you think?

Reply:
Ok, I did (2) for now (put it in respective llama2 and llama3 dirs). Let me know if this sounds reasonable to you

Reply:
Yeah I think this is reasonable. It is a bit of a pain to override all the necessary config fields to change models from the command line, so I think it makes sense to separate out QAT configs by model (since for (3) it'd get really verbose to change the model from the default config anyways). The downside is the configs are slightly less visible, but given we have a standalone top-level recipe this is OK imo.